File: crossecore-java/src/java_/CompilationUnit.java
package java_;
public interface CompilationUnit 
 extends NamedElement
{
	
	public static com.crossecore.ocl.QuickSet<CompilationUnit> allInstances_ = new com.crossecore.ocl.QuickSet<CompilationUnit>(CompilationUnit.class);
	
		
	public static com.crossecore.ocl.QuickSet<CompilationUnit> allInstances(){
		
		com.crossecore.ocl.QuickSet<CompilationUnit> result = new com.crossecore.ocl.QuickSet<CompilationUnit>(CompilationUnit.class);
		result.addAll(CompilationUnit.allInstances_);
		
		
		return result;
	}
	
	//public static com.crossecore.ocl.QuickSet<CompilationUnit> allInstances = new com.crossecore.ocl.QuickSet<CompilationUnit>(CompilationUnit.class);
	
	java.lang.String getOriginalFilePath();
	void setOriginalFilePath(java.lang.String value);
	com.crossecore.ocl.OrderedSet<Comment> getCommentList();
	com.crossecore.ocl.OrderedSet<ImportDeclaration> getImports();
	
	Package getPackage();
	void setPackage(Package value);
	com.crossecore.ocl.OrderedSet<AbstractTypeDeclaration> getTypes();
	
}

File: crossecore-typescript/java_/CompilationUnit.ts
import {OrderedSet} from "ecore/OrderedSet";
import {Set} from "ecore/Set";
import {Bag} from "ecore/Bag";
import {Comment} from "./Comment";
import {ImportDeclaration} from "./ImportDeclaration";
import {Package} from "./Package";
import {NamedElement} from "./NamedElement";
import {AbstractTypeDeclaration} from "./AbstractTypeDeclaration";

export interface CompilationUnit
extends NamedElement

{
	originalFilePath:string;
	
	commentList: OrderedSet<Comment>;
	imports: OrderedSet<ImportDeclaration>;
	package:Package;
	types: OrderedSet<AbstractTypeDeclaration>;
	

}

